docs/apache-airflow-providers-amazon/logging/s3-task-handler.rst

```diff
@@ -47,3 +47,86 @@ You can also use `LocalStack <https://localstack.cloud/>`_ to emulate Amazon S3
+
+Execute the following command by providing all the necessary inputs.
+
+.. code-block:: bash
+
+    eksctl create iamserviceaccount --cluster="<EKS_CLUSTER_ID>" --name="<SERVICE_ACCOUNT_NAME>" --namespace="<NAMESPACE>" --attach-policy-arn="<IAM_POLICY_ARN>" --approve``
+
+Example with sample inputs
+
+.. code-block:: bash
+
+    eksctl create iamserviceaccount --cluster=airflow-eks-cluster --name=airflow-sa --namespace=airflow --attach-policy-arn=arn:aws:iam::aws:policy/AmazonS3FullAccess --approve
+
+
+Step2: Update Helm Chart `values.yaml` with Service Account
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This step is using `Airflow Helm Chart <https://github.com/apache/airflow/tree/main/chart>`_ deployment.
+If you are deploying Airflow using Helm Chart then you can modify the ``values.yaml`` as mentioned below.
+Add the Service Account (e.g., ``airflow-sa``) created by Step1 to Helm Chart ``values.yaml`` under the following sections.
+We are  using the existing ``serviceAccount`` hence ``create: false`` with existing name as ``name: airflow-sa``.
+
+
+.. code-block:: yaml
+
+    workers:
+      serviceAccount:
+        create: false
+        name: airflow-sa
+        # Annotations are automatically added by **Step1** to serviceAccount. So, you dont need to mention the annotations. We have added this for information purpose
+        annotations:
+          eks.amazonaws.com/role-arn: <ENTER_IAM_ROLE_ARN_CREATED_BY_EKSCTL_COMMAND>
+
+    webserver:
+      serviceAccount:
+        create: false
+        name: airflow-sa
+        # Annotations are automatically added by **Step1** to serviceAccount. So, you dont need to mention the annotations. We have added this for information purpose
+        annotations:
+          eks.amazonaws.com/role-arn: <ENTER_IAM_ROLE_ARN_CREATED_BY_EKSCTL_COMMAND
+
+    config:
+      logging:
+        remote_logging: 'True'
+        logging_level: 'INFO'
+        remote_base_log_folder: 's3://<ENTER_YOUR_BUCKET_NAME>/<FOLDER_PATH' # Specify the S3 bucket used for logging
+        remote_log_conn_id: 'aws_s3_conn' # Notice that this name is used in Step3 for creating connections through Airflow UI
+        delete_worker_pods: 'False'
+        encrypt_s3_logs: 'True'
+
+Step3: Create Amazon S3 connection in Airflow Web UI
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+With the above configurations, Webserver and Worker Pods can access Amazon S3 bucket and write logs without using any Access Key and Secret Key or Instance profile credentials.
+
+The final step to create connections under Airflow UI before executing the DAGs
+
+* Login to Airflow Web UI with ``admin`` credentials and Navigate to ``Admin -> Connections``
+* Create connection for ``S3`` and select the options(Connection ID and Connection Type) as shown in the image.
```

Comment:
S3Hook, as well as all of hooks based on AwsBaseHook, intend to use [Amazon Web Services Connection](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html).

It could use Amazon S3 Connection (not documented, exists for historical reason???) but in this case users get UserWarning that connection has incorrect connection type.

Reply:
Thanks! It does make sense to use AWS base connection. I have updated the instructions and the image to reflect the Amazon Web Services base connection.